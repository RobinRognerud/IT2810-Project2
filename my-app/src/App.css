.App {
  text-align: center;
}

#header {
  height: 70px;
  width: 100%;
  background: grey;
}

.home-container {
  background: #aaa;
  height: 500px;
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: space-evenly;
}

audio {
  background-color: #aaa;

  text-indent: 0;

  border-radius: 5px;
}
