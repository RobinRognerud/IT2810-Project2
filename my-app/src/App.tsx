import React from "react";
import "./App.css";
import "./StartPage.css";
import Main from "./components/Main";
import Header from "./components/Header";
import Footer from "./components/Footer";
import { ThemeProvider, useTheme } from "./components/ThemeContext";

const helloRef = React.createRef<HTMLDivElement>();

const scroll = () => {
  if (helloRef && helloRef.current) {
    helloRef.current.scrollIntoView();
  }
};

function App() {
  return (
    <div className="App">
      <ThemeProvider>
        <div className="startPage">
          <h1 className="welcome-text">Welcome to our gallery</h1>
          {/* https://codepen.io/postor/pen/mskxI */}
          <div className="arrow-div">
            {" "}
            <svg className="arrows">
              <g onClick={scroll}>
                <path className="a1" d="M0 0 L30 32 L60 0"></path>
                <path className="a2" d="M0 20 L30 52 L60 20"></path>
                <path className="a3" d="M0 40 L30 72 L60 40"></path>
              </g>
            </svg>
          </div>
        </div>
        <Header />
        <div ref={helloRef}>
          <Main />
        </div>
        <div className="page-container">
          <DarkmodeButton />
        </div>
        <Footer />
      </ThemeProvider>
    </div>
  );
}

export default App;

const DarkmodeButton = () => {
  const { theme, setTheme } = useTheme();
  return (
    <div id="darkmodeButton-div">
      <button
        id="darkmodeButton"
        onClick={() => setTheme(theme === "dark" ? "light" : "dark")}
      >
        {theme === "dark" ? "Switch to light mode" : "Switch to dark mode"}
      </button>
    </div>
  );
};
