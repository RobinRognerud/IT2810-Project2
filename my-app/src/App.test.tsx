import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import App from "./App";

Enzyme.configure({ adapter: new Adapter() });

it("App loads main correctly", () => {
  const wrapper = Enzyme.mount(<App />);
  const main = wrapper.find("Main");
  expect(main).toBeTruthy();
});

it("App loads darkmodeButton correctly", () => {
  const wrapper = Enzyme.mount(<App />);
  const darkmodeButton = wrapper.find("DarkmodeButton");
  expect(darkmodeButton).toBeTruthy();
});

it("App loads footer correctly", () => {
  const wrapper = Enzyme.mount(<App />);
  const footer = wrapper.find("Footer");
  expect(footer).toBeTruthy();
});

describe("App", () => {
  test("loads without error", () => {
    let element = <App />;

    expect(element).toBeTruthy();
  });
});

describe("App", () => {
  test("render matches snapshot", () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
