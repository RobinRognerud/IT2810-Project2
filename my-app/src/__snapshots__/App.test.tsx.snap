// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="startPage"
  >
    <h1
      className="welcome-text"
    >
      Welcome to our gallery
    </h1>
    <div
      className="arrow-div"
    >
       
      <svg
        className="arrows"
      >
        <g
          onClick={[Function]}
        >
          <path
            className="a1"
            d="M0 0 L30 32 L60 0"
          />
          <path
            className="a2"
            d="M0 20 L30 52 L60 20"
          />
          <path
            className="a3"
            d="M0 40 L30 72 L60 40"
          />
        </g>
      </svg>
    </div>
  </div>
  <div>
    <div
      className="home-container"
      id="home"
    >
      <div
        className="picture-container"
      >
        <div
          id="product-options"
        >
          <div
            id="picture-category"
          >
            <svg
              fill="none"
              id="singer"
              viewBox="0 0 888 773"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                id="undraw_happy_music_g6wc 1"
              >
                <g
                  id="shape"
                >
                  <path
                    d="M442.277 669.71C534.082 669.71 608.505 595.287 608.505 503.482C608.505 411.677 534.082 337.254 442.277 337.254C350.472 337.254 276.049 411.677 276.049 503.482C276.049 595.287 350.472 669.71 442.277 669.71Z"
                    fill="#3F3D56"
                    id="Vector"
                  />
                </g>
                <g
                  id="eyes"
                >
                  <path
                    d="M337.626 516.826C341.754 511.883 346.918 507.908 352.751 505.179C358.584 502.451 364.946 501.037 371.385 501.037C377.825 501.037 384.187 502.451 390.02 505.179C395.853 507.908 401.017 511.883 405.145 516.826C412.209 510.082 417.092 501.377 419.163 491.833C421.235 482.289 420.401 472.343 416.769 463.278C413.136 454.212 406.872 446.442 398.783 440.969C390.694 435.496 381.152 432.571 371.385 432.571C361.619 432.571 352.077 435.496 343.988 440.969C335.899 446.442 329.635 454.212 326.002 463.278C322.37 472.343 321.536 482.289 323.607 491.833C325.679 501.377 330.562 510.082 337.626 516.826V516.826Z"
                    fill="white"
                    id="Vector_3"
                  />
                  <path
                    d="M469.631 516.826C473.759 511.883 478.922 507.908 484.756 505.179C490.589 502.451 496.95 501.037 503.39 501.037C509.83 501.037 516.191 502.451 522.025 505.179C527.858 507.908 533.021 511.883 537.15 516.826C544.214 510.082 549.096 501.377 551.168 491.833C553.24 482.289 552.406 472.343 548.773 463.278C545.141 454.212 538.877 446.442 530.788 440.969C522.699 435.496 513.157 432.571 503.39 432.571C493.624 432.571 484.081 435.496 475.993 440.969C467.904 446.442 461.64 454.212 458.007 463.278C454.375 472.343 453.541 482.289 455.612 491.833C457.684 501.377 462.567 510.082 469.631 516.826Z"
                    fill="white"
                    id="Vector_4"
                  />
                  <path
                    d="M354.707 481.632C363.996 481.632 371.525 474.102 371.525 464.814C371.525 455.526 363.996 447.996 354.707 447.996C345.419 447.996 337.889 455.526 337.889 464.814C337.889 474.102 345.419 481.632 354.707 481.632Z"
                    fill="#3F3D56"
                    id="Vector_5"
                  />
                  <path
                    d="M486.707 481.632C495.996 481.632 503.525 474.102 503.525 464.814C503.525 455.526 495.996 447.996 486.707 447.996C477.419 447.996 469.889 455.526 469.889 464.814C469.889 474.102 477.419 481.632 486.707 481.632Z"
                    fill="#3F3D56"
                    id="Vector_6"
                  />
                </g>
                <g
                  id="earphone"
                >
                  <path
                    d="M286.25 566.475C279.83 566.475 273.472 565.21 267.541 562.754C261.609 560.297 256.219 556.695 251.679 552.155C247.14 547.615 243.538 542.226 241.081 536.294C238.624 530.362 237.36 524.005 237.36 517.584V493.139C237.36 480.172 242.511 467.737 251.679 458.568C260.848 449.399 273.284 444.248 286.25 444.248V444.248V566.475Z"
                    fill="#6C63FF"
                    id="Vector_9"
                  />
                  <path
                    d="M598.303 444.248C604.724 444.248 611.081 445.513 617.013 447.97C622.945 450.427 628.334 454.028 632.874 458.568C637.414 463.108 641.016 468.498 643.473 474.429C645.93 480.361 647.194 486.719 647.194 493.139V517.584C647.194 524.005 645.93 530.362 643.473 536.294C641.016 542.226 637.414 547.615 632.874 552.155C628.334 556.695 622.945 560.296 617.013 562.753C611.081 565.21 604.724 566.475 598.304 566.475V566.475V444.248H598.303Z"
                    fill="#6C63FF"
                    id="Vector_10"
                  />
                  <path
                    d="M623.923 496.98H616.166C616.166 398.603 536.13 318.567 437.752 318.567C339.375 318.567 259.338 398.603 259.338 496.98H251.581C251.581 394.325 335.097 310.81 437.752 310.81C540.408 310.81 623.923 394.326 623.923 496.98Z"
                    fill="#6C63FF"
                    id="Vector_11"
                  />
                </g>
                <g
                  id="music1"
                >
                  <path
                    d="M171.547 347.473L171.549 347.47L167.904 345.953L152.528 339.452L147.016 337.123L146.593 338.19L139.189 356.884L160.026 365.704L160.136 365.803L158.731 369.114C158.539 369.416 158.375 369.736 158.241 370.069C157.258 372.553 159.116 375.023 160.872 375.718C162.628 376.414 164.849 374.965 165.833 372.481C166.803 370.031 165.855 366.919 164.143 366.167L164.145 366.164L160.501 364.647L148.034 359.8L155.157 341.815L154.101 341.397L146.975 359.389L145.434 358.79L152.588 340.728L152.776 340.807L167.43 347.01L167.539 347.109L166.135 350.42C165.943 350.722 165.779 351.042 165.645 351.375C164.662 353.859 166.519 356.329 168.276 357.024C170.032 357.72 172.253 356.271 173.237 353.787C174.207 351.337 173.259 348.225 171.547 347.473Z"
                    fill="#3F3D56"
                    id="Vector_12"
                  />
                  <path
                    d="M243.667 393.367C236.103 389.043 237.617 379.552 237.617 379.552C237.426 373.387 232.333 374.26 232.333 374.26L230.479 374.683C229.618 374.283 228.669 374.108 227.722 374.173L224.902 374.648C231.324 376.156 230.796 381.528 230.796 381.528L231.696 387.809L214.87 381.021L216.802 376.466C217.069 376.046 217.297 375.602 217.482 375.139C218.848 371.69 216.268 368.259 213.829 367.293C211.39 366.327 208.305 368.34 206.939 371.789C205.591 375.192 206.908 379.514 209.285 380.558L209.283 380.562L214.345 382.67L214.382 382.551L243.667 393.367ZM235.123 382.331L235.662 388.343C233.757 385.67 232.862 382.408 233.137 379.137C233.092 377.973 232.699 376.85 232.01 375.911C233.07 376.591 233.922 377.55 234.471 378.683C235.02 379.816 235.246 381.078 235.123 382.331V382.331Z"
                    fill="#3F3D56"
                    id="Vector_13"
                  />
                  <path
                    d="M192.766 277.541L190.816 282.14C190.549 282.56 190.321 283.004 190.135 283.467C188.769 286.916 191.349 290.347 193.788 291.313C196.228 292.279 199.313 290.266 200.679 286.817C202.026 283.414 200.709 279.092 198.332 278.048L198.334 278.044L193.273 275.936L175.958 269.205L185.851 244.225L184.385 243.644L174.488 268.633L172.347 267.801L182.282 242.714L182.544 242.825L202.897 251.439L203.049 251.577L201.099 256.175C200.832 256.596 200.604 257.04 200.418 257.503C199.052 260.952 201.632 264.383 204.072 265.349C206.511 266.315 209.596 264.302 210.962 260.853C212.309 257.45 210.992 253.128 208.616 252.083L208.618 252.079L203.556 249.972L182.871 241.227L182.87 241.227L182.199 240.943L174.544 237.707L163.674 265.155L192.614 277.404L192.766 277.541Z"
                    fill="#3F3D56"
                    id="Vector_14"
                  />
                </g>
                <g
                  id="music2"
                >
                  <path
                    d="M115.319 304.456L115.321 304.452L110.259 302.344L89.1049 294.12L94.4441 289.37C95.4886 287.978 97.0311 287.043 98.7488 286.762C100.467 286.48 102.227 286.874 103.661 287.86L102.004 285.398C102.004 285.398 98.8892 281.274 94.5291 285.637C94.5291 285.637 89.1342 293.59 80.6603 291.563L109.601 303.812L109.752 303.95L107.802 308.548C107.535 308.968 107.307 309.413 107.122 309.875C105.756 313.324 108.336 316.755 110.775 317.721C113.214 318.688 116.299 316.674 117.665 313.225C119.013 309.823 117.696 305.5 115.319 304.456Z"
                    fill="#3F3D56"
                    id="Vector_15"
                  />
                  <path
                    d="M101.054 136.972L101.057 136.965L92.3439 133.336L55.9277 119.18L65.1188 111.003C65.1188 111.003 71.2325 102.52 80.985 108.404L78.133 104.164C78.133 104.164 72.7708 97.0656 65.2652 104.576C65.2652 104.576 55.9783 118.267 41.3909 114.777L91.2098 135.863L91.4713 136.1L88.1142 144.016C87.6547 144.74 87.2626 145.505 86.9429 146.3C84.5912 152.238 89.0328 158.145 93.2316 159.808C97.4305 161.471 102.741 158.005 105.093 152.067C107.412 146.21 105.145 138.77 101.054 136.972Z"
                    fill="#3F3D56"
                    id="Vector_16"
                  />
                  <path
                    d="M127.794 255.885C109.709 245.546 113.328 222.855 113.328 222.855C112.873 208.114 100.694 210.201 100.694 210.201L96.2614 211.213C94.2024 210.257 91.9352 209.838 89.6706 209.994L82.9281 211.128C98.2811 214.736 97.0197 227.579 97.0197 227.579L99.1706 242.596L58.9416 226.367L63.5605 215.476C64.1989 214.471 64.7435 213.409 65.1875 212.303C68.4538 204.056 62.2848 195.853 56.4529 193.543C50.621 191.233 43.2454 196.047 39.9791 204.294C36.7573 212.428 39.9059 222.763 45.5889 225.26L45.5841 225.27L57.6859 230.309L57.776 230.024L127.794 255.885ZM107.365 229.499L108.659 243.923C101.063 233.84 102.616 221.861 102.616 221.861C102.509 219.079 101.57 216.393 99.9217 214.15C108.647 219.371 107.364 229.499 107.364 229.499H107.365Z"
                    fill="#3F3D56"
                    id="Vector_17"
                  />
                </g>
                <g
                  id="mouth"
                >
                  <path
                    d="M486.146 546.219C487.89 552.345 488.411 558.754 487.678 565.081C486.945 571.407 484.973 577.528 481.875 583.093C478.777 588.657 474.613 593.557 469.621 597.513C464.629 601.468 458.907 604.402 452.781 606.146C446.655 607.89 440.246 608.411 433.919 607.678C427.593 606.945 421.472 604.973 415.907 601.875C410.343 598.777 405.443 594.613 401.487 589.621C397.532 584.629 394.598 578.907 392.854 572.781L439.5 559.5L486.146 546.219Z"
                    fill="#FF6584"
                    id="Ellipse 1"
                  />
                  <path
                    d="M461.14 553.339C461.949 556.18 462.191 559.154 461.851 562.089C461.511 565.024 460.596 567.863 459.158 570.445C457.721 573.027 455.789 575.3 453.474 577.135C451.158 578.97 448.503 580.331 445.661 581.14C442.82 581.949 439.846 582.191 436.911 581.851C433.976 581.511 431.137 580.596 428.555 579.158C425.973 577.721 423.7 575.789 421.865 573.474C420.03 571.158 418.669 568.503 417.86 565.661L439.5 559.5L461.14 553.339Z"
                    fill="#E83559"
                    id="Ellipse 2"
                  />
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="text-container"
      >
        <div
          className="poetry"
        >
          <div>
            <p
              id="title"
            >
               
              
            </p>
          </div>
          <div>
             
            <p>
              
            </p>
             
          </div>
          <div>
            <p
              id="author"
            >
               Written by 
              
            </p>
          </div>
        </div>
        <div
          className="music"
        >
          <audio
            controls={true}
            src="happyBall.mp3"
          />
        </div>
      </div>
      <div
        className="options-container"
      >
        <p
          id="option-text"
        >
           
          Make your own art by changing image, music and poetry:
        </p>
        <div
          id="product-options"
        >
          <div
            id="picture-category"
          >
            <p
              id="titleRB"
            >
              Picture:
            </p>
            <label>
              Singing ball
              <input
                checked={true}
                className="radio"
                name="pictureNr"
                onChange={[Function]}
                type="radio"
                value="0"
              />
            </label>
            <label>
              Abstract
              <input
                checked={false}
                className="radio"
                name="pictureNr"
                onChange={[Function]}
                type="radio"
                value="1"
              />
            </label>
            <label>
              Tree
              <input
                checked={false}
                className="radio"
                name="pictureNr"
                onChange={[Function]}
                type="radio"
                value="2"
              />
            </label>
          </div>
        </div>
        <div
          id="product-options"
        >
          <div
            id="music-category"
          >
            <p
              id="titleRB"
            >
              Music:
            </p>
            <label>
              Happy
              <input
                checked={true}
                className="radio"
                name="musicNr"
                onChange={[Function]}
                type="radio"
                value="0"
              />
            </label>
            <label>
              Powerful
              <input
                checked={false}
                className="radio"
                name="musicNr"
                onChange={[Function]}
                type="radio"
                value="1"
              />
            </label>
            <label>
              Wind
              <input
                checked={false}
                className="radio"
                name="musicNr"
                onChange={[Function]}
                type="radio"
                value="2"
              />
            </label>
          </div>
        </div>
        <div
          id="product-options"
        >
          <div
            id="poetry-category"
          >
            <p
              id="titleRB"
            >
               Poetry:
            </p>
            <label>
              Dance
              <input
                checked={true}
                className="radio"
                name="poetryNr"
                onChange={[Function]}
                type="radio"
                value="0"
              />
            </label>
            <label>
              The star
              <input
                checked={false}
                className="radio"
                name="poetryNr"
                onChange={[Function]}
                type="radio"
                value="1"
              />
            </label>
            <label>
              Summer
              <input
                checked={false}
                className="radio"
                name="poetryNr"
                onChange={[Function]}
                type="radio"
                value="2"
              />
            </label>
          </div>
        </div>
        <p
          id="favorite-text"
        >
          Save your favorite art:
        </p>
        <div
          className="FavoriteButton-container"
        >
          <button
            className="favoriteButton"
            onClick={[Function]}
          >
            Save favorite ❤
          </button>
        </div>
        <div
          className="FavoriteButton-container"
        >
          <button
            className="favoriteButton"
            onClick={[Function]}
          >
            Get favorite ❤
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="darkmode-container"
  >
    <div
      id="darkmodeButton-div"
    >
      <button
        id="darkmodeButton"
        onClick={[Function]}
      >
        Switch to dark mode
      </button>
    </div>
  </div>
  <footer>
    <div
      className="footer"
    >
      <p />
    </div>
  </footer>
</div>
`;
